tests/integration/api_server/conftest.py

```diff
@@ -0,0 +1,52 @@
+# pylint: disable=redefined-outer-name
+import time
+import pytest
+import bentoml
+
+from .example_service import gen_test_bundle
+
+
+@pytest.fixture(params=[True, False], scope="session")
+def enable_microbatch(request):
+    pytest.enable_microbatch = request.param
+    return request.param
+
+
+@pytest.fixture(autouse=True, scope='session')
+def image(tmpdir_factory):
+    import docker
+
+    client = docker.from_env()
+
+    bundle_dir = tmpdir_factory.mktemp('test_bundle')
+    bundle_path = str(bundle_dir)
+    gen_test_bundle(bundle_path)
+    image = client.images.build(path=bundle_path, tag="example_service", rm=True)[0]
+    yield image
+    client.images.remove(image.id)
+
+
+@pytest.fixture(autouse=True, scope='session')
+def host(image, enable_microbatch):
+    import docker
+
+    client = docker.from_env()
+
+    with bentoml.utils.reserve_free_port() as port:
+        pass
+    if enable_microbatch:
+        command = "bentoml serve-gunicorn /bento --enable-microbatch --workers 1"
+    else:
+        command = "bentoml serve-gunicorn /bento --workers 1"
+    container = client.containers.run(
+        command=command,
+        image=image.id,
+        auto_remove=True,
+        tty=True,
+        ports={'5000/tcp': port},
+        detach=True,
+    )
+    time.sleep(10)
```

Comment:
can we use the container logs to tell that the container is ready here?

Reply:
```
➜  BentoML bentoml serve-gunicorn /Users/agent/bentoml/repository/BentoSvc/20200706164630_B92371 --port 5000 --workers 2
[2020-07-10 00:19:54 +0800] [39546] [INFO] Starting gunicorn 20.0.4
[2020-07-10 00:19:54 +0800] [39546] [INFO] Listening at: http://0.0.0.0:5000 (39546)
[2020-07-10 00:19:54 +0800] [39546] [INFO] Using worker: sync
[2020-07-10 00:19:54 +0800] [39560] [INFO] Booting worker with pid: 39560
[2020-07-10 00:19:54 +0800] [39566] [INFO] Booting worker with pid: 39566
[2020-07-10 00:19:55,236] WARNING - Using BentoML installed in `editable` model, the local BentoML repository including all code changes will be packaged together with saved bundle created, under the './bundled_pip_dependencies' directory of the saved bundle.
[2020-07-10 00:19:55,260] WARNING - Saved BentoService bundle version mismatch: loading BentoService bundle create with BentoML version 0.8.1, but loading from BentoML version 0.8.3+13.g87462db
[2020-07-10 00:19:55,283] WARNING - bentoml.handlers.* will be deprecated after BentoML 1.0, use bentoml.adapters.* instead
[2020-07-10 00:19:55,285] WARNING - DataframeHandler will be deprecated after BentoML 1.0, use DataframeInput instead
[2020-07-10 00:19:55,307] WARNING - Using BentoML installed in `editable` model, the local BentoML repository including all code changes will be packaged together with saved bundle created, under the './bundled_pip_dependencies' directory of the saved bundle.
[2020-07-10 00:19:55,343] WARNING - Saved BentoService bundle version mismatch: loading BentoService bundle create with BentoML version 0.8.1, but loading from BentoML version 0.8.3+13.g87462db
[2020-07-10 00:19:55,365] WARNING - bentoml.handlers.* will be deprecated after BentoML 1.0, use bentoml.adapters.* instead
[2020-07-10 00:19:55,367] WARNING - DataframeHandler will be deprecated after BentoML 1.0, use DataframeInput instead
```

Reply:
@parano 
I didn't find a suitable sign. Maybe wait for `GET /healthz` getting ready?